### DISCUSSION TITLE
When using the ngram approach, how to boost the ngram field?

### DISCUSSION THREAD

**Discussion by amenk** [Smile-SA/elasticsuite#2574] [Q&A]
I created a custom module based on https://github.com/Smile-SA/elasticsuite/issues/2543#issuecomment-1114689369

This basically works pretty well, but we want to optimize the relevance.

Ngram results should weight similarly high like normal matches in the title.
Currently we get matches from the description weighted higher then ngram matches in the title.

Do I have to manipulate that query or can I do it somewhere else? 

![image](https://user-images.githubusercontent.com/1087128/170650562-d2ba596c-f824-4358-86c0-7bbed5f39e6d.png)

The problem is, that there is no setter for the fields array.

As a Proof of concept I patched:

\Smile\ElasticsuiteCore\Index\Mapping::getWeightedSearchProperties

            if ($property && $canAddField) {
                if ($property == 'name.ngram') {
                    $weightedFields[$property] = 80;
                } else {
                $weightedFields[$property] = $boost * $field->getSearchWeight();
                }
            }

Which basically works, but I am wondering if there is a nicer way to boost this name.ngram field.

**Comment by florisschreuder** [comment]
I don't have a solution, but I'm upvoting the idea for an easily extensible function to boost 'field.analyzer' combinations.

Possibly with an extensible array within di.xml ? Would require extremely little changes and is very easy to add into.

**Comment by amenk** [comment]
Current solution (still feel *hacky*)

```
<?php

namespace IMI\ElasticsuiteNgram\Plugin\Index;

class MappingPlugin
{
    public function afterGetWeightedSearchProperties(\Smile\ElasticsuiteCore\Index\Mapping $subject, $result)
    {
        if (isset($result['name.ngram'])) {
            $result['name.ngram'] = 70;
        }

        return $result;
    }
}
```
